### ISSUE TITLE
Client IP preservation fails with Windows + SLB and externalTrafficPolicy:Local

### ISSUE THREAD

**Issue by brobichaud** [Azure/AKS#1157]
**What happened**:
Running a new AKS cluster with k8s 1.14.5 and using a Standard Load Balancer. When creating a  service that specifies Local for externalTrafficPolicy my Windows pods are not seeing the true client ip address. Instead they are seeing the ip address of any one of the 4 nodes in my cluster (Linux and Windows nodes).

**What you expected to happen**:
Pods to see the true client ip on incoming requests. This issue indicates it should work: https://github.com/Azure/AKS/issues/1028

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a new cluster that uses a Standard Load Balancer
2. Viewing the details of the cluster using the "az aks show" command results in:
```
"networkProfile": {
    "dnsServiceIp": "10.0.0.10",
    "dockerBridgeCidr": "172.17.0.1/16",
    "loadBalancerSku": "standard",
    "networkPlugin": "azure",
    "networkPolicy": null,
    "podCidr": null,
    "serviceCidr": "10.0.0.0/16"
  },
```
3. Create a k8s service that specifies Local for externalTrafficPolicy
4. Viewing the details of the service using the "kubectl get svc" command results in:
```
Name:                     dev-clientip-svc
Namespace:                default
Labels:                   component=clientip
                          environment=dev
Annotations:              kubectl.kubernetes.io/last-applied-configuration:
                            {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"component":"clientip","environment":"dev"},"name":"dev-clienti...
Selector:                 component=clientip,environment=dev
Type:                     LoadBalancer
IP:                       10.0.191.127
LoadBalancer Ingress:     <obscured>
Port:                     https  443/TCP
TargetPort:               44372/TCP
NodePort:                 https  31762/TCP
Endpoints:                10.240.0.107:44372,10.240.0.130:44372
Session Affinity:         None
External Traffic Policy:  Local
HealthCheck NodePort:     30525
```
5. Create a deployment with labels that match the selectors of the above service
6. Call a method on that deployment that captures and returns the client ip in the response

**Anything else we need to know?**:
The client ip I see in my pod matches any one of the 4 cluster node ip addresses.

**Environment**:
- Kubernetes version: 1.14.5
- Size of cluster: 2 nodepools, the default linux nodepool with 2 nodes, and a Windows nodepool with 2 nodes.
- General description of workloads in the cluster: AspNetCore v2.2 webapi
- Others:

**Comment by brobichaud** [comment]
Any thoughts on this? It's a significant roadblock for us in containerizing some of our workloads as we use IP Whitelisting for some key services. Everything I read says this should work with a Standard Load Balancer, but it does not seem to.

**Comment by bitsofinfo** [comment]
update please? how can we see the external client IP making a request to a service in aks?

**Comment by jnoller** [comment]
- Source IP for normal services exposed via type loadBalancer works.
- Users must set externalTrafficPolicy=Local
- If the user is not using an Ingress Controller, SSL pass-through works
- Source IP for services using ingress controllers (all):
  - SSL pass-through with ingress controllers will not work, this would be a man in the middle attack and break SSL
    - This is a known and accepted upstream limitation
If the user is using ingress controllers but not attempting SSL pass through, source IP preservation will work
    - Users should NOT set use-proxy-protocol: "true" (Azure LBs (SLB, BLB) do not support this)
    - Users must use the “X-Forwarded-For” header as the source of truth to get information about the client IP address

**Comment by brobichaud** [comment]
@jnoller as this issue describes, client IP preservation is NOT working as you indicate for me.

**Comment by jnoller** [comment]
Ok, @brobichaud - thanks for clarifying, Has a support ticket been opened so we can access the cluster? We will need to do that as this has been confirmed to work, so this may be specific to your clusters configuration or a potential regression

**Comment by palma21** [comment]
Hi @brobichaud I can only repro this with Windows is that your understanding as well, or did you find any issues with Linux nodes?

**Comment by brobichaud** [comment]
@palma21 yes, I'm running Windows workloads so am only seeing this in Windows pods. I have not tried with linux pods.

**Comment by brobichaud** [comment]
@jnoller I had not yet opened a support ticket as I was thinking for preview features this was the place to start. I've gone ahead and created ticket #119092626002914 that references this github issue.

**Comment by jluk** [comment]
Thanks @brobichaud we'll follow on through the support ticket but leave this open for tracking eventual resolution for Windows preview. FYI @MikkelHegn

**Comment by MikkelHegn** [comment]
Does anyone know if this is Windows specific or any node pool beyond the default first node pool? (If not I can go and verify)

**Comment by palma21** [comment]
AFAIK It's only windows specific. Can't repro on Linux only.

**Comment by brobichaud** [comment]
This remains a significant roadblock for migrating all of our workloads to AKS. Knowing the incoming requests Client IP address is a pretty fundamental feature in some contexts.

We have now moved all of our workloads to AKS in production form, with the GLARING exception of any workload that utilizes or relies upon client ip addresses. We are completely roadblocked on those.

Is there any way I can advocate for this feature being addressed soon?

**Comment by MikkelHegn** [comment]
Let me add the upstream bug tracking this. There's a dependency on moving to newer version of Windows beyond the Kubernetes work. https://github.com/kubernetes/kubernetes/issues/62046.

**Comment by brobichaud** [comment]
Nothing has changed and this remains a significant roadblock for migrating all of our workloads to AKS.

Knowing the incoming requests Client IP address is a pretty fundamental feature in some contexts. We have now moved all of our workloads to AKS in production form, with the glaring exception of any workload that relies upon client ip addresses. We are completely roadblocked on those.

I understand that this may be resolved in k8s v1.20 but it feels like this ought to remain open until that is conclusive and released.
